easybuild/easyblocks/l/lammps.py

```diff
@@ -0,0 +1,353 @@
+            print_warning(warning_msg)
+
+            cpu_arch = self.get_cpu_arch()
+
+            if cpu_arch == "sandybridge" or cpu_arch == "ivybridge":
+                processor_arch = 'SNB'
+            elif cpu_arch == "haswell":
+                processor_arch = 'HSW'
+            elif cpu_arch == "broadwell":
+                processor_arch = 'BDW'
+            elif cpu_arch == "skylake":
+                processor_arch = 'SKX'
+            elif cpu_arch == "knights-landing":
+                processor_arch = 'KNL'
+            else:
+                error_msg = "Couldn't determine CPU architecture, you need to set 'kokkos_arch' manually."
+                raise EasyBuildError(error_msg)
+                exit(1)
+            print("Determined cpu arch: %s" % processor_arch)
+
+        if not cuda:
+            return processor_arch
+
+        # CUDA below
+        cuda_cc.sort(reverse=True)
+        gpu_arch = None
+        for cc in cuda_cc:
+            gpu_arch = KOKKOS_GPU_ARCH_TABLE.get(str(cc))
+            if gpu_arch:
+                break
+            else:
+                warning_msg = "(%s) GPU ARCH was not found in listed options." % cc
+                print_warning(warning_msg)
+
+        if not gpu_arch:
+            error_msg = "Specified GPU ARCH (%s) " % cuda_cc
+            error_msg += "was not found in listed options [%s]." % KOKKOS_GPU_ARCH_TABLE
+            raise EasyBuildError(error_msg)
+        return "%s;%s" % (processor_arch, gpu_arch)
+
+    def check_cuda_compute_capabilities(self, cfg_cuda_cc, ec_cuda_cc, cuda_cc):
+        cuda = get_software_root('CUDA')
+
+        if cuda:
+            if cfg_cuda_cc and ec_cuda_cc:
+                warning_msg = "cuda_compute_capabilities specified in easyconfig (%s)" % self.ec_cuda_cc
+                warning_msg += " are overruled by "
+                warning_msg += "--cuda-compute-capabilities configuration option (%s)" % self.cfg_cuda_cc
+                print_warning(warning_msg)
+            elif not cuda_cc:
+                error_msg = "No CUDA compute capabilities specified.\nTo build LAMMPS with Cuda you need to use"
+                error_msg += "the --cuda-compute-capabilities configuration option or the cuda_compute_capabilities "
+                error_msg += "easyconfig parameter to specify a list of CUDA compute capabilities to compile with."
+                raise EasyBuildError(error_msg)
+
+        elif cuda_cc:
+            warning_msg = "Missing CUDA package (in dependencies), "
+            warning_msg += "but 'cuda_compute_capabilities' option was specified."
+            print_warning(warning_msg)
+
+        return cuda_cc
+
+    def get_cpu_arch(self):
+        out, ec = run_cmd("python -c 'from archspec.cpu import host; print(host())'", simple=False)
+        if ec:
+            raise EasyBuildError("Failed to determine CPU architecture: %s", out)
+        # transform: 'skylake_avx512\n' => 'skylake'
```

Comment:
I wouldn't mangle the output obtained from `archspec`.

`skylake` is actually "Skylake desktop", which doesn't have AVX512. So the Skylake expected by LAMMPS (to use `SKX` is actually `skylake_avx512` (sometimes called "Skylake X")

Reply:
oh, didn't know that :) thanks